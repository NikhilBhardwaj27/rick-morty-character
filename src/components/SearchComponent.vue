<template>
  <div class="mt-4">
    <p class="ml-3">Search By Name</p>
    <input
      type="text"
      class="w-full p-2 border-2 border-black ml-3 lg:w-[400px]"
      aria-label="search-box"
    />
    <button
      class="rounded bg-slate-400 hidden text-white p-[11px] ml-2 lg:inline-block"
    >
      Search
    </button>
  </div>
</template>
<script setup></script>
