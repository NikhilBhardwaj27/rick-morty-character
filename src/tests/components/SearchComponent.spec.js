import {mount} from "@vue/test-utils";
import SearchComponent from '../../components/SearchComponent.vue';

const wrapper = mount(SearchComponent);

it("snapshot UI testing of SearchComponent.vue", async () => {
    expect(wrapper.text()).toMatchSnapshot()
});

it("Render Heading Search by Name", async () => {
    const text = wrapper.find('p')
    expect(text.text()).toContain('Search By Name')
});


