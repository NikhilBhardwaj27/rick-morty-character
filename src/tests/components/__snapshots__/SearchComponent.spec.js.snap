// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot UI testing of SearchComponent.vue 1`] = `"Search By Name Search"`;
